src/components/Table/Table.js

```diff
@@ -0,0 +1,122 @@
+import _ from "lodash";
+import { MDCRipple } from "@material/ripple";
+
+import tableTemplate from "./Table.html";
+import { DotsMenuComponent } from "../DotsMenu/DotsMenu";
+
+export class TableComponent {
+  constructor(mountPoint, props = {}) {
+    this.mountPoint = mountPoint;
+    this.props = props;
+    this.state = {
+      data: this.props.data,
+      initialData: Array.from(this.props.data),
+      orderTypes: ["initial", "asc", "desc"],
+      currentOrderTypeIndex: 1,
+      columnName: ""
+    };
+
+    this.handleOrderClick = this.handleOrderClick.bind(this);
+  }
+
+  getColumnName(target) {
+    if (target.tagName === "BUTTON") {
+      return target.firstElementChild.attributes.data.value;
+    }
+    if (target.tagName === "svg") {
+      return target.attributes.data.value;
+    }
+    return null;
+  }
+
+  handleOrderClick(e) {
+    const { data, initialData, orderTypes, currentOrderTypeIndex } = this.state;
+
+    this.state.columnName = this.getColumnName(e.target);
+    if (!this.state.columnName) {
+      return;
+    }
+
+    if (currentOrderTypeIndex === 0) {
+      this.state.data = initialData;
+    } else {
+      this.state.data = _.orderBy(
+        data,
+        this.state.columnName,
+        orderTypes[currentOrderTypeIndex]
+      );
+    }
+
+    this.mount();
+  }
+
+  querySelectors() {
+    const { mountPoint } = this;
+    this.tableHead = mountPoint.querySelector(".table__head");
+    this.iconButtonRipple = mountPoint.querySelectorAll(".mdc-icon-button");
```

Comment:
please do not query material class names

Reply:
done